pkg/kubectl/cmd/util/factory.go

```diff
@@ -1305,3 +1256,68 @@ func (f *Factory) NewBuilder(thirdPartyDiscovery bool) *resource.Builder {
 
 	return resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true))
 }
+
+// useDiscoveryRESTMapper checks the server version to see if its recent enough to have
+// enough discovery information available to reliably build a RESTMapper.  If not, use the
+// hardcoded mapper in this client (legacy behavior)
+func useDiscoveryRESTMapper(serverVersion string) bool {
```

Comment:
Discovery information was incomplete before.  I agree that this will end up causing a problem when running against servers that either don't host `/version` or reflect their versions, not kube versions.

I'll open a followup issue.